### PR TITLE
chore: add compose profiles for web modeler

### PR DESCRIPTION
**Pull request by JacobiCamunda**
Needed to setup for a follow-up PR which implements: https://github.com/camunda/web-modeler/issues/9437

This PR introduces [compose profiles](https://docs.docker.com/compose/profiles/) to get rid of `docker-compose-web-modeler.yaml` and set up for the issue mentioned above.

### CODE REVIEW

**Comment by JacobiCamunda** [comment]
I found a strange bug: If I run the full suite with Web Modeler (`docker compose --profile modeler up -d`) and open Web Modeler first I get the following error when trying to access Identity after:

```
2024-05-27 11:51:15 io.camunda.identity.sdk.impl.rest.exception.RestException: request failed with status code '400' and body '{"error":"invalid_grant","error_description":"Invalid refresh token"}'
2024-05-27 11:51:15     at io.camunda.identity.sdk.impl.rest.RestClient.send(RestClient.java:123) ~[identity-sdk-8.5.0-SNAPSHOT.jar!/:8.5.0-SNAPSHOT]
2024-05-27 11:51:15     at io.camunda.identity.sdk.impl.rest.RestClient.request(RestClient.java:105) ~[identity-sdk-8.5.0-SNAPSHOT.jar!/:8.5.0-SNAPSHOT]
2024-05-27 11:51:15     at io.camunda.identity.sdk.impl.keycloak.KeycloakAuthentication.renewToken(KeycloakAuthentication.java:105) ~[identity-sdk-8.5.0-SNAPSHOT.jar!/:8.5.0-SNAPSHOT]
2024-05-27 11:51:15     at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[na:na]
2024-05-27 11:51:15     at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(Unknown Source) ~[na:na]
2024-05-27 11:51:15     at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source) ~[na:na]
2024-05-27 11:51:15     at java.base/java.lang.reflect.Method.invoke(Unknown Source) ~[na:na]
2024-05-27 11:51:15     at io.camunda.identity.sdk.annotation.AnnotationProcessor.lambda$apply$0(AnnotationProcessor.java:33) ~[identity-sdk-8.5.0-SNAPSHOT.jar!/:8.5.0-SNAPSHOT]
```

Accessing Identity first and Web Modeler after is working fine. Doing the same on the main branch with `docker compose -f docker-compose.yaml -f docker-compose-web-modeler.yaml up -d` is working fine in both cases.

Therefore this change must introduce this but I cannot find the root cause.

**Comment by wollefitz** [comment]
> I found a strange bug: If I run the full suite with Web Modeler (`docker compose --profile modeler up -d`) and open Web Modeler first I get the following error when trying to access Identity after:

Hey @JacobiCamunda! I just had a quick look and can't reproduce - for me, everything works fine when starting via the command you shared.

File: docker-compose.yaml [hunk 1]
```diff
@@ -373,9 +509,11 @@ volumes:
   kibana:
   operate_tmp:
   tasklist_tmp:
+  postgres-web:
 
 networks:
-  # Note there are two bridge networks: One for Camunda Platform and one for Identity.
-  # Identity and Keycloak are part of both as they need to be accessible by platform components.
+  # Note there are three bridge networks: One for Camunda Platform, one for Identity and one for Web Modeler.
+  # Identity and Keycloak are part of the first two as they need to be accessible by platform components.
```
Comment (wollefitz) [inline docker-compose.yaml:516]: Let's also explain why certain Web modeler components are part of other networks here.

File: docker-compose.yaml [hunk 2]
```diff
@@ -373,9 +509,11 @@ volumes:
...
   camunda-platform:
   identity-network:
+  modeler:
```
Comment (wollefitz) [inline docker-compose.yaml:519]: 💭 Do we want to call this `web-modeler` instead?

File: README.md [hunk 3]
```diff
@@ -163,20 +163,20 @@ $ docker login registry.camunda.cloud
 Username: your_username
 Password: ******
 Login Succeeded
-$ docker compose -f docker-compose.yaml -f docker-compose-web-modeler.yaml up -d
+$ docker compose --profile modeler up -d
```
Comment (wollefitz) [inline README.md:166]: 💭 Do we want to call this profile `web-modeler` instead?

File: docker-compose.yaml (hunk 2 above)
Comment (JacobiCamunda) [reply docker-compose.yaml:519]: Should we rename every "modeler" occurrence to "web-modeler" too, then? For example, the service name `modeler-websocket` -> `web-modeler-websockets`?

Comment (wollefitz) [reply docker-compose.yaml]: Yes, let's do that 👍

**Review by wollefitz — COMMENTED** [review]
Lgtm

The disadvantage about renaming Web Modeler's containers from `modeler-*` to `web-modeler-*` is that when you start the stack with the state before this PR and switch to the new version without tearing down the environment first, you'll get conflicts as docker-compose won't make the link between the containers with the old name.

I'll leave the final review to @camunda/distribution

**Pull request merged**